pkg/database/realm.go

```diff
@@ -1981,17 +2002,17 @@ func (db *Database) AllRealmCodeStats(ctx context.Context, requiredRealms int, e
 		) d
 		LEFT JOIN realm_stats s ON s.date = d.date
 		`
-	if realm != nil {
+	if len(excludedRealmIDs) > 0 {
```

Comment:
This should always be appended to the query, and the values appended to the values list. SQL will optimize away a `NOT INT []` call to be a noop search.